com.unity.render-pipelines.universal/Shaders/Utils/Deferred.hlsl

```diff
@@ -44,6 +44,9 @@ struct PunctualLightData
     int flags;               // Light flags (enum kLightFlags and LightFlag in C# code)
     half4 occlusionProbeInfo;
     int shadowLightIndex;
+    int spare0;
```

Comment:
Do you know if we can define packing rules for this? Otherwise we will have to rename this vars if we give meaning to them later and that could be a considered a breaking change.

Reply:
I have a new version of this file without the `spare0` member. I didn't submit the new code because I was waiting for this PR to get through since it already passed all Yamato tests.
Generally, I don't think URP has an API to declare the fields of the struct with correct packing/alignment, has it? Such API would also need to know how to handle half types as storage.

Reply:
I pushed a new version without the `spare0` field.
I promoted back some fields from half to float, as half in structured buffer is considered 16bits (which is not the case in cbuffer on non-metal platforms). Eventually, we need an API to declare shared structures between C# code and shader code that deal with automatic alignment and half storage types.
All these changes will not affect stencil deferred. I locally tested UniversalGraphicsTest with forward and deferred renderer and all tests are passing as usual.